lib/improver/cli/generate_orographic_alphas.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI for generating orographic alphas."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(orography: cli.inputcube,
+            *,
+            min_alpha: float = 0.0,
+            max_alpha: float = 1.0,
+            coefficient: float = 1.0,
+            power: float = 1.0,
+            invert_alphas=True):
+    """Generate alpha smoothing parameters for recursive filtering based on
+    orography gradients.
+
+    Args:
+        orography (iris.cube.Cube):
+            A 2D field of orography for the grid to generate alphas for.
+        min_alpha (float):
+            The minimum value of alpha.
+        max_alpha (float):
+            The maximum value of alpha.
+        coefficient (float):
+            The coefficient for the alpha calculation.
+        power (float):
+            The power for the alpha equation.
+        invert_alphas (bool):
+            If True then the max and min alpha values will be swapped.
```

Comment:
@bayliffe, what does this do and why is this needed?

Reply:
In the called plugin the user defines min and max alphas between which the returned alpha values are scaled. The steepest orography gradient will be associated with the minimum alpha in the default case (invert=True, as set in the plugin interface). If the user instead wants to associate the maximum alpha with the steepest gradient they can use invert_alphas=False.

The same can be achieved by simply reversing the min and max values that the users provides (e.g. min_value = 0.5, max_value = 0) though in that case we should rename them as they make little sense with max value < min value.

Reply:
(I thought it was probably possible to remove this by renaming the arguments - since this isn't used in the suite we can change this without affecting suite testing etc)

Reply:
Yes, we should eliminate `invert_alphas` if all it does is to swap around `min_alpha` and `max_alpha`.

Reply:
`max_grad_alpha`, `low_grad_alpha`?